File: src/test/testData/java/implicit/HashMap.java
package ru.ustinov;

import java.net.URL;
import java.util.HashMap;
import java.util.HashSet;

class Simple {
    public static void main() {
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">HashMap<URL, Integer> hashMap;</warning>
        HashMap<Integer, URL> hashMap2;
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">HashMap<HashSet<URL>, Integer> hashMap3;</warning>
        HashMap<HashSet<Integer>, URL> hashMap6;
    }
}

File: src/test/testData/kotlin/implicit/HashMap.kt
import java.net.URL
import java.util.HashSet

@Suppress("UNUSED_VARIABLE")
fun main() {
<warning descr="This data structure can probably implicitly invoke URL#hashCode()">val a: HashSet<URL></warning>
        val b: HashSet<Int>
<warning descr="This data structure can probably implicitly invoke URL#hashCode()">val c: HashSet<HashSet<URL>></warning>
        val d: HashSet<HashSet<Int>>
        }